execution/gethexec/sequencer.go

```diff
@@ -1658,19 +1658,25 @@ func (s *Sequencer) Start(ctxIn context.Context) error {
 		})
 	}
 
-	s.CallIteratively(func(ctx context.Context) time.Duration {
-		nextBlock := time.Now().Add(s.config().MaxBlockSpeed)
-		if s.createBlock(ctx) {
-			// Note: this may return a negative duration, but timers are fine with that (they treat negative durations as 0).
-			return time.Until(nextBlock)
-		}
-		// If we didn't make a block, try again immediately.
-		return 0
-	})
-
 	return nil
 }
 
+func (s *Sequencer) StartSequencing(ctx context.Context) (*execution.SequencedMsg, time.Duration) {
+	sequencedMsg, err := s.execEngine.SequenceDelayedMessage()
+	if err != nil {
+		return nil, 0
+	}
+	if sequencedMsg != nil {
+		return sequencedMsg, 0
+	}
+
+	sequencedMsg, waitUntilSequencingNextBlock := s.createBlockWithRegularTxs(ctx)
```

Comment:
Maybe I'm missing some context, but isn't there a possibility to starve `createBlockWithRegularTxs()` if Sequencer keeps getting delayed messages? Or not a problem here?